sambaplayersdk/src/main/java/com/sambatech/player/SambaPlayer.java

```diff
@@ -649,6 +670,11 @@ public float getCurrentTime() {
      * @return Float total duration
      */
     public float getDuration() {
+
+        if (sambaCast != null && sambaCast.isCasting() && castPlayer != null) {
+            return castPlayer.getDuration() / 1000f;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1000f to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.sambatech.com/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a "double" or "BigDecimal" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.sambatech.com/coding_rules#rule_key=squid%3AS2164)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.sambatech.com/coding_rules#rule_key=squid%3AS818)